app/models/bus_stops_route.rb

```diff
@@ -9,4 +9,45 @@ class BusStopsRoute < ApplicationRecord
   validates :direction, presence: true
   validates_uniqueness_of :sequence, scope: %i[route direction]
   validates_uniqueness_of :bus_stop, scope: %i[route direction]
+
+  def self.import(stop_hash)
+    stop_hash.each_pair do |route_dir, direction_variants|
+      variants = direction_variants.values
+      longest_variant = variants.max_by(&:length)
+      other_variants = variants - longest_variant
+      other_variants.each do |other_variant|
+        other_variant.each.with_index do |stop, sequence|
+          unless longest_variant.include? stop
+            # if sequence is zero, then it is the first stop. We can't rely on
+            # its previous stop like we do in the corresponding else statement.
+            if sequence == 0
+              # first stop on longest variant that's on the other variant
+              first_known_stop = longest_variant.find do |known_stop|
+                other_variant.include? known_stop
+              end
+              if first_known_stop.present?
+                # insert stop before first known stop on longest variant
+                first_known_stop_index = longest_variant.index(first_known_stop)
+                longest_variant.insert(first_known_stop_index, stop)
+              # the other variant has no stops in common with the longest variant
+              else
+                # add the stop onto the end of the longest variant
+                longest_variant << stop
+              end
+            else
+              # the previous stop will already be in the longest variant
+              previous_stop = other_variant[sequence - 1]
+              previous_stop_index_in_longest_variant = longest_variant.index(previous_stop)
+              # Array#insert inserts before the given index.
+              longest_variant.insert(previous_stop_index_in_longest_variant + 1, stop)
```

Comment:
Feel free to disagree on my verbose variable names.

Reply:
I think they're fine since they are descriptive.